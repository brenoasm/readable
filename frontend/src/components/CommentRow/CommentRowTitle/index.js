import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import EditIcon from '../../icons/EditIcon';
import DeleteIcon from '../../icons/DeleteIcon';

const propTypes = {};

const defaultProps = {};

const StyledCommentRowTitle = styled.div`
  display: flex;

  span:first-child {
    font-weight: 600;
  }

  div {
    margin-left: auto;

    span {
      margin-left: 10px;
    }
  }
`;

const CommentRowTitle = ({
  comment,
  editClick,
  deleteComment
}) => (
  <StyledCommentRowTitle>
    <span>{comment.author}</span>
    <div>
      <EditIcon title="Editar comentário" handleEdit={() => editClick(comment)} />
      <DeleteIcon title="Remover comentário" handleDelete={() => deleteComment(comment.id)} />
    </div>
  </StyledCommentRowTitle>
);

CommentRowTitle.propTypes = propTypes;
CommentRowTitle.defaultProps = defaultProps;

export default CommentRowTitle;
