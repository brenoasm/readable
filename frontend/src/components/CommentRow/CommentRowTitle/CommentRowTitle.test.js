/* eslint-disable */

import React from 'react';

import CommentRowTitle from './';

describe('CommentRowTitle', () => {
  let wrapper;

  beforeEach(() => {
    const initialState = {
      comment: {
        id: '1',
        parentId: '12',
        timestamp: 1538417732043,
        body: 'Test',
        author: 'Breno',
        voteScore: 1,
        deleted: false,
        parentDeleted: false
      },
      editClick: jest.fn(),
      deleteComment: jest.fn()
    };

    wrapper = shallow(<CommentRowTitle {...initialState} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have one EditIcon component', () => {
    expect(wrapper.find('EditIcon')).toHaveLength(1);
  });

  it('should have one DeleteIcon component', () => {
    expect(wrapper.find('DeleteIcon')).toHaveLength(1);
  });
});
