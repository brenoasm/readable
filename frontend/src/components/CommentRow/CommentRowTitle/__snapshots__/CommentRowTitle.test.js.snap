// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentRowTitle matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentRowTitle
    comment={
        Object {
            "author": "Breno",
            "body": "Test",
            "deleted": false,
            "id": "1",
            "parentDeleted": false,
            "parentId": "12",
            "timestamp": 1538417732043,
            "voteScore": 1,
          }
    }
    deleteComment={[Function]}
    editClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <span>
          Breno
</span>,
        <div>
          <EditIcon
                    handleEdit={[Function]}
                    objectToEdit={Object {}}
                    title="Editar comentário"
          />
          <DeleteIcon
                    handleDelete={[Function]}
                    objectToDelete={Object {}}
                    title="Remover comentário"
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Breno",
        },
        "ref": null,
        "rendered": "Breno",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <EditIcon
              handleEdit={[Function]}
              objectToEdit={Object {}}
              title="Editar comentário"
/>,
            <DeleteIcon
              handleDelete={[Function]}
              objectToDelete={Object {}}
              title="Remover comentário"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleEdit": [Function],
              "objectToEdit": Object {},
              "title": "Editar comentário",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleDelete": [Function],
              "objectToDelete": Object {},
              "title": "Remover comentário",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <span>
            Breno
</span>,
          <div>
            <EditIcon
                        handleEdit={[Function]}
                        objectToEdit={Object {}}
                        title="Editar comentário"
            />
            <DeleteIcon
                        handleDelete={[Function]}
                        objectToDelete={Object {}}
                        title="Remover comentário"
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Breno",
          },
          "ref": null,
          "rendered": "Breno",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <EditIcon
                handleEdit={[Function]}
                objectToEdit={Object {}}
                title="Editar comentário"
/>,
              <DeleteIcon
                handleDelete={[Function]}
                objectToDelete={Object {}}
                title="Remover comentário"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleEdit": [Function],
                "objectToEdit": Object {},
                "title": "Editar comentário",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleDelete": [Function],
                "objectToDelete": Object {},
                "title": "Remover comentário",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
